error.go

```
@@ -96,6 +96,9 @@ func (e *restError) Unwrap() error {
 //
 //	if err != nil {return restful.NewError(nil, http.StatusBadRequest, "bad data")}
 func NewError(err error, statusCode int, description ...string) error {
+	if _, ok := err.(*restError); ok {
```

You silently drop the provided arguments. That is misleading.